### PR TITLE
created product cards, page

### PR DESCRIPTION
**Pull request by Ksenaaa**

### CODE REVIEW

File: src/components/Checkbox/Checkbox.module.scss [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+    border: 1px solid $lightBlueTransparent;
+    transition: $animationTransition;
+  }
+
+  & input:hover ~ .checkmark {
+    background: $lightBlueTransparent;
+  }
+
+  & input:checked ~ .checkmark {
+    background: url(./../../img/svg/checkbox.svg) center;
```
Comment (vell28) [inline src/components/Checkbox/Checkbox.module.scss]: try to use global path pls

File: src/components/Main/Testimonials/Testimonials.tsx [hunk 2]
```diff
@@ -3,55 +3,44 @@ import React from 'react'
...
-      </div>
-    </div>
+    <CommonSlider
+      titleName="Testimonials"
+      pathTitle={`/${Routes.TESTIMONIALS}`}
+      sliderLength={testimonials.length}
+    >
+      {testimonials?.map(item =>
+        <CommonSliderItem key={item.id}>
+          <Link href="/" className={styles.wrapper}>
```
Comment (vell28) [inline src/components/Main/Testimonials/Testimonials.tsx]: add route to variable

File: src/components/MoreInformation/MoreInformation.tsx [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+  onToggle: () => void
+}
+
+export const MoreInformation = ({ minHeight, onToggle, children }: PropsWithChildren<Props>) => {
+  const [height, setHeight] = useState(0)
+
+  const ref = useRef<HTMLDivElement | null>(null) as MutableRefObject<HTMLDivElement>
+
+  useEffect(() => {
+    setHeight(ref.current.clientHeight)
```
Comment (vell28) [inline src/components/MoreInformation/MoreInformation.tsx]: do we really need set new value for state, every time when we open/close ?

File: src/components/ProductCard/ProductCard.tsx [hunk 4]
```diff
@@ -0,0 +1,90 @@
...
+  const { data, isSuccess } = useGetColorsQuery()
+  const colors = isSuccess ? productColors(product.colors, data) : []
+
+  const getBlur = useCallback((img: string) => getBlurImage(img), [])
+
+  const { t } = useTranslation()
+  const categories = t('menu', { returnObjects: true }) as IMenuName[]
+
+  const handlerClickLike = (e: MouseEvent) => {
+    e.preventDefault()
```
Comment (vell28) [inline src/components/ProductCard/ProductCard.tsx]: move it to utils and use every where, and add TODO or comment, whan is this.

File: src/components/ProductCard/ProductCard.tsx [hunk 5]
```diff
@@ -0,0 +1,90 @@
...
+          alt="img"
+          fill
+          sizes="33vw"
+          placeholder="blur"
+          blurDataURL={getBlur(product.img)}
+        />
+      </div>
+      <div className={styles.categoryWithCode}>
+        <div className={styles.category}>
+          {categories?.find(item => item.id === product.category)?.name}
```
Comment (vell28) [inline src/components/ProductCard/ProductCard.tsx]: don't do this inside render, move it above and add to variable

File: src/components/ProductInfo/ProductCharacteristics/PriceSelection/PriceSelection.tsx [hunk 6]
```diff
@@ -0,0 +1,42 @@
...
+  return (
+    <div className={styles.priceSelection} onClick={handlerChangePrice}>
+      {parsePrices?.map(category =>
+        <div className={styles.priceCategory} key={category.name}>
+          <Checkbox
+            type="radio"
+            labelName={category.name}
+            radioName="price"
+            value={category.value}
+            defaultChecked={category.name === parsePrices[0].name && true}
```
Comment (vell28) [inline src/components/ProductInfo/ProductCharacteristics/PriceSelection/PriceSelection.tsx]: `category.name === parsePrices[0].name && true` 
i sure that is not correct

File: src/components/ProductInfo/ProductCharacteristics/ProductCharacteristics.tsx [hunk 7]
```diff
@@ -0,0 +1,125 @@
...
+  }
+
+  const onAddProduct = () => {
+    setAmountProduct(preState => preState + 1)
+  }
+
+  const { data: dataColors, isSuccess } = useGetColorsQuery()
+  const colors = isSuccess ? productColors(product.colors, dataColors) : []
+
+  const productMaterials = [...new Set(colors.map(i => i.material))]
```
Comment (vell28) [inline src/components/ProductInfo/ProductCharacteristics/ProductCharacteristics.tsx]: what is `i`

File: src/components/ProductInfo/ProductCharacteristics/ProductCharacteristics.tsx [hunk 8]
```diff
@@ -0,0 +1,125 @@
...
+          <Button name="Add to cart" />
+        </div>
+        <div className={styles.like} onClick={handlerClickLike}>
+          <Image src={likeIcon} alt="like" fill />
+        </div>
+      </div>
+
+      <PriceSelection
+        product={product}
+        onChangePrice={onChangePrice}
```
Comment (vell28) [inline src/components/ProductInfo/ProductCharacteristics/ProductCharacteristics.tsx]: why we can't pass setPrice insted onChangePrice?

File: src/components/ProductInfo/ProductInfo.tsx [hunk 9]
```diff
@@ -0,0 +1,58 @@
...
+export const ProductInfo: FC<Props> = ({ id }) => {
+  const { data, isSuccess } = useGetProductQuery(id)
+
+  const product = isSuccess ? data : {} as IProduct
+
+  const { isOpen: isOpenTextAbout, onToggle: toggleTextAbout } = useToggle(true)
+
+  return (
+    <div className={styles.wrapper}>
+      {isSuccess &&
```
Comment (vell28) [inline src/components/ProductInfo/ProductInfo.tsx]: if `isSuccess === false` we will render blank `div` i guess it's nut correct

File: src/components/ProductInfo/ProductPictures/ProductPictures.tsx [hunk 10]
```diff
@@ -0,0 +1,68 @@
...
+  }, [])
+
+  return (
+    <div className={styles.pictures}>
+      <div className={styles.otherPictures}>
+        <Slider
+          picturesLength={pictures.length}
+          isAutoCarousel={false}
+          isSwipe={true}
+          axisScroll={windowWidth <= 680 ? AxisScroll.X : AxisScroll.Y}
```
Comment (vell28) [inline src/components/ProductInfo/ProductPictures/ProductPictures.tsx]: what is `680`? magic variable

File: src/components/ProductInfo/ProductPictures/ProductPictures.tsx [hunk 11]
```diff
@@ -0,0 +1,68 @@
...
+          column={windowWidth <= 680 ? 3 : 6}
```
Comment (vell28) [inline src/components/ProductInfo/ProductPictures/ProductPictures.tsx]: 3 and 6 ? magic

File: src/interface/productInterface.ts [hunk 12]
```diff
@@ -0,0 +1,42 @@
+export interface IProduct {
+  id: string,
+  name: string,
+  code: string,
+  category: string,
+  type: string,
+  popularity: number,
+  quality: {
```
Comment (vell28) [inline src/interface/productInterface.ts]: we tolk about `quantity` interface, but i don't see it

File: src/store/services/product.ts [hunk 13]
```diff
@@ -0,0 +1,40 @@
...
+        url: 'product/new'
+      }),
+      providesTags: () => [{ type: 'Product' }]
+    }),
+    getAllProductsByCategory: builder.query<IPageData<ICardProduct[]>, string>({
+      query: category => `product/${category}`,
+      providesTags: (arg) => [{ type: 'Product', category: arg }]
+    }),
+    getSomeProductsByCategory: builder.query<ICardProduct[], string>({
+      query: category => `product/some-category/${category}`,
```
Comment (vell28) [inline src/store/services/product.ts]: `some-category`? it's sound very strange

File: src/store/services/product.ts [hunk 14]
```diff
@@ -0,0 +1,40 @@
...
+      providesTags: (arg) => [{ type: 'Product', category: arg }]
+    }),
+    getSomeProductsByType: builder.query<ICardProduct[], string>({
+      query: typeProduct => `product/some-type/${typeProduct}`,
```
Comment (vell28) [inline src/store/services/product.ts]: the same

**Review by vell28 — APPROVED** [review]

File: src/components/Main/Testimonials/Testimonials.tsx (hunk 2 above)
Comment (Ksenaaa) [reply src/components/Main/Testimonials/Testimonials.tsx]: added a route to the Testimonials page. Created a task - will scroll to this element on the Testimonials page

File: src/components/ProductCard/ProductCard.tsx (hunk 4 above)
Comment (Ksenaaa) [reply src/components/ProductCard/ProductCard.tsx]: deleted, will be used in the future

File: src/components/ProductInfo/ProductCharacteristics/PriceSelection/PriceSelection.tsx (hunk 6 above)
Comment (Ksenaaa) [reply src/components/ProductInfo/ProductCharacteristics/PriceSelection/PriceSelection.tsx]: fixed

File: src/components/ProductInfo/ProductCharacteristics/ProductCharacteristics.tsx (hunk 7 above)
Comment (Ksenaaa) [reply src/components/ProductInfo/ProductCharacteristics/ProductCharacteristics.tsx]: fixed

File: src/components/ProductInfo/ProductCharacteristics/ProductCharacteristics.tsx (hunk 8 above)
Comment (Ksenaaa) [reply src/components/ProductInfo/ProductCharacteristics/ProductCharacteristics.tsx]: fixed

File: src/components/ProductInfo/ProductInfo.tsx (hunk 9 above)
Comment (Ksenaaa) [reply src/components/ProductInfo/ProductInfo.tsx]: added loader

File: src/store/services/product.ts (hunk 13 above)
Comment (Ksenaaa) [reply src/store/services/product.ts]: fixed

File: src/store/services/product.ts (hunk 14 above)
Comment (Ksenaaa) [reply src/store/services/product.ts]: fixed

File: src/components/ProductCard/ProductCard.tsx (hunk 5 above)
Comment (Ksenaaa) [reply src/components/ProductCard/ProductCard.tsx]: fixed

File: src/components/Checkbox/Checkbox.module.scss (hunk 1 above)
Comment (Ksenaaa) [reply src/components/Checkbox/Checkbox.module.scss]: fixed

File: src/components/MoreInformation/MoreInformation.tsx (hunk 3 above)
Comment (Ksenaaa) [reply src/components/MoreInformation/MoreInformation.tsx]: fixed

File: src/components/ProductInfo/ProductPictures/ProductPictures.tsx (hunk 10 above)
Comment (Ksenaaa) [reply src/components/ProductInfo/ProductPictures/ProductPictures.tsx]: fixed

File: src/components/ProductInfo/ProductPictures/ProductPictures.tsx (hunk 11 above)
Comment (Ksenaaa) [reply src/components/ProductInfo/ProductPictures/ProductPictures.tsx]: fixed

File: src/interface/productInterface.ts (hunk 12 above)
Comment (Ksenaaa) [reply src/interface/productInterface.ts]: fixed

File: src/components/CommonSlider/CommonSlider.tsx [hunk 15]
```diff
@@ -0,0 +1,54 @@
...
+
+import styles from './CommonSlider.module.scss'
+
+type Props = {
+  titleName: string,
+  pathTitle?: string,
+  sliderLength: number,
+}
+
+export const CommonSlider = ({ titleName, pathTitle, sliderLength, children }: PropsWithChildren<Props>) => {
```
Comment (vell28) [inline src/components/CommonSlider/CommonSlider.tsx:22]: CommonSlider:  PropsWithChildren<Props>

Comment (Ksenaaa) [reply src/components/CommonSlider/CommonSlider.tsx:22]: discussed

**Pull request merged**